WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/ProductRowViewModel.swift

```diff
@@ -174,8 +174,14 @@ private extension ProductRowViewModel {
     }
 }
 
-extension ProductRowViewModel {
+extension ProductRowViewModel: Equatable {
     static func == (lhs: ProductRowViewModel, rhs: ProductRowViewModel) -> Bool {
-        lhs.id == rhs.id
+        return lhs.productID == rhs.productID &&
+        lhs.imageURL == rhs.imageURL &&
+        lhs.name == rhs.name &&
+        lhs.stockAndPriceLabel == rhs.stockAndPriceLabel &&
+        lhs.skuLabel == rhs.skuLabel &&
+        lhs.quantity == rhs.quantity &&
+        lhs.canChangeQuantity == rhs.canChangeQuantity
```

Comment:
I wonder if we should group all or most of these properties on a `struct`, something like `ProductDetails`.
And then only compare that product details type, that will prevent us on forgetting to update the custom equality when updating the view model.

Reply:
Shouldn't we be comparing the generated `id` for any scenario? Is it not needed on the new order screen?